index.html

```diff
@@ -244,9 +269,71 @@
         </dt>
       </dl>
       <p>
-        The <code>PermissionName</code> enum defines the list of known
-        permission names.
+        Each enumerator-value in the <a>PermissionName</a> enum identifies a
+        <dfn data-export="">permission</dfn>, which consists of the following
+        algorithms and types:
       </p>
+      <dl>
+        <dt>
+          An <dfn data-export="">associated PermissionDescriptor</dfn>
+        </dt>
+        <dd>
+          A subtype of <a>PermissionDescriptor</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission storage type</dfn>
+        </dt>
+        <dd>
+          A subtype of <a>PermissionStorage</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission result type</dfn>
+        </dt>
+        <dd>
+          A subtype of <a>PermissionStatus</a>.
+        </dd>
+        <dt>
+          An algorithm to <dfn data-export="">get a permission storage
+          identifier</dfn>.
+        </dt>
+        <dd>
+          Takes an <a>environment settings object</a> argument and returns a
+          <a>permission storage identifier</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission query algorithm</dfn>
+        </dt>
+        <dd>
+          Takes an <a>associated PermissionDescriptor</a>, an instance of the
+          <a>permission storage type</a> that's currently stored for this
+          <a>permission</a>, and an existing instance of the <a>permission
+          result type</a>, and updates the <a>permission result type</a>
+          instance with the query result. Used by <a for="Permissions" lt=
+          "query">Permissions.query()</a> and the <a href=
+          "#PermissionStatus-update">PermissionStatus update steps</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission request algorithm</dfn>
+        </dt>
+        <dd>
+          Takes the previously-stored <a>permission storage type</a>, an
+          <a>associated PermissionDescriptor</a>, and a newly-created
+          <a>permission result type</a>, shows the user any necessary prompt to
+          try to increase permissions, and updates the <a>permission storage
+          type</a> and <a>permission result type</a> to match. Used by <a for=
+          "Permissions" lt="request">Permissions.request()</a>, which handles
+          reading and writing the <a>permission store</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission revocation algorithm</dfn>
+        </dt>
+        <dd>
+          After <a for="Permissions" lt="revoke">Permissions.revoke()</a>
```

Comment:
This paragraph is inconsistent with the other two algorithms. Should be something like "Takes no arguments. Updates any other parts of the implementation that need to be kept in sync after an entry is removed from the permission store. Triggered by Permissions's revoke() method.